internal/bgp/bgp.go

```diff
@@ -665,3 +679,30 @@ func buildTCPMD5Sig(addr net.IP, key string) tcpmd5sig {
 
 	return t
 }
+
+// localAddressExists returns true if the provided address exists locally on
+// any of the host's network interfaces. The function doesn't return errors and
+// instead returns false if an error has occurred while looking for the
+// address. This function is safe to call if addr is nil or malformed.
+func localAddressExists(addr string) bool {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return false
+	}
+
```

Comment:
You probably want to check that the interface is up. See this function, for example: https://github.com/metallb/metallb/blob/80bfe1cf45bf4338e6eced11ab7a88d3d2356663/internal/layer2/announcer.go#L46

Reply:
You can use an address that's configured on an interface that is down as the source address :smile: Weird, but it works on my testing, at least for the outbound packets. I have no idea if a BGP session will actually be established in case the user uses an address of a downed interface, however I don't think we should actively prevent users from trying to do so.

Reply:
What you mean with "you can use"? What will happen? The same as with any address not configured on the server?